File: parse_pdf.py
import PyPDF2

def extract_text():
    try:
        reader = PyPDF2.PdfReader('algeria_69_provinces_communes.pdf')
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        
        with open('raw_pdf_text.txt', 'w', encoding='utf-8') as f:
            f.write(text)
        print("Successfully extracted PDF text to raw_pdf_text.txt")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    extract_text()

File: parse_pdf.js
const fs = require('fs');
const PDFParser = require("pdf2json");

async function extractWilayas() {
    try {
        const pdfParser = new PDFParser(null, 1);

        pdfParser.on("pdfParser_dataError", errData => console.error(errData.parserError));
        pdfParser.on("pdfParser_dataReady", pdfData => {
            const text = pdfParser.getRawTextContent();
            fs.writeFileSync('raw_pdf_text.txt', text);
            console.log("Extracted raw text. Please examine raw_pdf_text.txt to adjust regex.");
        });

        pdfParser.loadPDF("algeria_69_provinces_communes.pdf");
    } catch (e) {
        console.error("Error reading PDF:", e);
    }
}

extractWilayas();
